src/vs/code/electron-main/menubar.ts

```diff
@@ -277,6 +252,15 @@ export class Menubar {
 			menubar.append(gotoMenuItem);
 		}
 
+		// Terminal
```

Comment:
It does not feel right that this component is aware of the terminal menu.
Couldn't this be done fully from the terminal.contribution? What is missing for that?

Reply:
do you specifically mean for terminal, tasks and preferences? or are you referring to every top-level menu?

Reply:
I guess the nicest would be that every top level menu item is registered outside and that we do not reference them manually here